### ISSUE TITLE
Normalize MARL objective by number of agents

### ISSUE THREAD

**Issue by jdrotleff** [SwarmRL/SwarmRL#130]
Current implementation computes per-agent returns, but aggregates the actor and critic losses by summation. Consequently, the gradient magnitude scales with the number of particles.

The following files currently sum over particles:
- swarmrl/losses/proximal_policy_loss.py
    - entropy aggregation
    - critic loss aggregation
    - actor loss aggregation

- swarmrl/losses/policy_gradient_loss.py
    - critic loss aggregation
    - actor loss aggregation

This corresponds approximately to:
$L_{\text{current}}=\sum_i L_i$
rather than:
$L_{\text{MARL}}=\frac{1}{N}\sum_i L_i.$

### Proposed change

```
critic_loss = optax.huber_loss(
    predicted_values, returns
).sum(axis=0).mean()

actor_loss = clipped_loss.sum(axis=0).mean()

entropy = self.sampling_strategy.compute_entropy(
    new_probabilities
).sum(axis=0).mean()
```

 ### Why this matters
 Using a particle sum makes the effective update magnitude depend on the number of particles. This means
  - changing the number of agents changes the effective learning rate
  - experiments with different swarm sizes are not directly comparable
  - larger swarms may produce disproportionately large updates
